persistence/src/main/java/org/cbioportal/genome_nexus/persistence/SignalMutationRepository.java

```diff
@@ -1,12 +1,16 @@
 package org.cbioportal.genome_nexus.persistence;
 
 import org.cbioportal.genome_nexus.model.SignalMutation;
+import org.springframework.cache.annotation.Cacheable;
 import org.springframework.data.mongodb.repository.MongoRepository;
 
 import java.util.List;
 
 public interface SignalMutationRepository extends MongoRepository<SignalMutation, String>
 {
+    @Cacheable("allSignalMutations")
+    List<SignalMutation> findAll();
```

Comment:
This keeps all the mutations in memory. If this is not desired, we can process the mutations and only keep the search fields in another partial model. Alternatively, we can perform custom DB queries instead of pulling all mutations at once.